Userland/Libraries/LibJS/Runtime/ArrayBufferPrototype.cpp

```diff
@@ -66,12 +66,48 @@ static ArrayBuffer* array_buffer_object_from(VM& vm, GlobalObject& global_object
     return static_cast<ArrayBuffer*>(&this_object);
 }
 
+// 25.1.5.3 ArrayBuffer.prototype.slice, https://tc39.es/ecma262/#sec-arraybuffer.prototype.slice
 JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::slice)
 {
+    const auto start = vm.argument(0);
+    const auto end = vm.argument(1);
+
     auto array_buffer_object = array_buffer_object_from(vm, global_object);
     if (!array_buffer_object)
         return {};
-    TODO();
+
+    // FIXME: Check for shared buffer
+    // FIXME: Check for detached buffer
+
+    const auto len = array_buffer_object->byte_length();
+
+    const auto relative_start = start.is_negative_infinity()
+        ? 0
+        : start.to_integer_or_infinity(global_object);
+    const auto first = relative_start < 0
+        ? AK::max(len + relative_start, double(0))
+        : AK::min(relative_start, double(len));
+
+    const auto relative_end = end.is_undefined()
+        ? len
+        : end.to_integer_or_infinity(global_object);
+
+    double final;
+    if (end.is_negative_infinity())
+        final = 0;
+    else if (relative_end < 0)
+        final = AK::max(len + relative_end, double(0));
+    else
+        final = AK::min(relative_end, double(len));
+
+    const auto new_len = AK::max(final - first, double(0));
+
+    auto buffer = ArrayBuffer::create(global_object, new_len);
+
+    auto sliced = array_buffer_object->buffer().slice(first, new_len);
+    buffer->buffer() = sliced;
```

Comment:
Please add an appropriate setter to `ArrayBuffer` instead (if you choose to implement the additional steps above), or a constructor that takes a `ByteBuffer` instead of a `size_t` so we don't have to create one twice.